pydeseq2/dds.py

```diff
@@ -227,14 +230,34 @@ def deseq2(self) -> None:
             # for genes that had outliers replaced
             self.refit()
 
-    def fit_size_factors(self) -> None:
+    def fit_size_factors(
+        self, fit_type: Literal["ratio", "iterative"] = "ratio"
+    ) -> None:
         """Fit sample-wise deseq2 normalization (size) factors.
 
-        Uses the median-of-ratios method: see :func:`pydeseq2.preprocessing.deseq2_norm`.
+        Uses the median-of-ratios method: see :func:`pydeseq2.preprocessing.deseq2_norm`,
+        unless each gene has at least one zero, in which case it switches to
+        the ``iterative`` method.
+
+        Parameters
+        ----------
+        fit_type : str
+            The normalization method to use (default: ``"ratio"``).
         """
         print("Fitting size factors...")
         start = time.time()
-        self.layers["normed_counts"], self.obsm["size_factors"] = deseq2_norm(self.X)
+        if fit_type == "iterative":
+            self._fit_iterate_size_factors()
+        # Test whether median it is possible to use median-of-ratios.
+        elif (self.X == 0).any(0).all():
+            # There is at least a zero for each gene
+            print(
+                "Every gene contains at least one zero, "
+                "cannot compute log geometric means. Switching to iterative mode."
```

Comment:
```suggestion
                "All genes contain at least one sample with a zero count, "
                "cannot compute log geometric means. Switching to iterative mode."
```

Reply:
Do you think this could be better a warning than just a print?

Reply:
Changed to RuntimeWarnign